apps/vscode/command_client/command_client_tag.py

```diff
@@ -0,0 +1,23 @@
+import string
+
+from talon import Module
+
+mod = Module()
+
+mod.tag(
+    "command_client", desc="For applications which implement File based RPC with Talon"
+)
+
+
+@mod.action_class
+class Actions:
+    def command_server_directory() -> string:
+        """The dirctory which contains the files required for communication between the application and Talon.
+        This is the only function which absolutely must be implemented for any application using the command-client."""
+
+    def emit_pre_phrase_signal() -> bool:
+        """The command client can touch a signal file at the start of a phrase. If your
+        implementation does not require this, override emit_pre_phrase_signal to
+        return False."""
+        # Unless we're in a command client app, we do nothing.
+        return False
```

Comment:
I don't really understand why this is a separate file from `command_client.py`, or how to determine what goes in which file.

Reply:
I thought that all tags should reside in there own file. I put the Actions here because these are the actions a user command_client would need to implement as shown in the visual_studio.py file. I thought putting them in command_client would make it more difficult to assertain what needed to be implemeted for another system. I'm definitly a Newbie when it comes to Python and Talon scripting, so any guidence whould be appreciated.